gcsio/src/test/java/com/google/cloud/hadoop/gcsio/TrackingHttpRequestInitializer.java

```diff
@@ -169,12 +171,33 @@ public ImmutableList<String> getAllRequestStrings() {
         .collect(toImmutableList());
   }
 
+  public ImmutableList<String> getAllRequestInvocationIds() {
+    return requests.stream()
+        .map(r -> getInvocationId(r.getHeaders()))
+        .filter(id -> id != null && !id.isEmpty())
+        .collect(toImmutableList());
+  }
+
   public ImmutableList<String> getAllRawRequestStrings() {
     return requests.stream()
         .map(GoogleCloudStorageIntegrationHelper::requestToString)
         .collect(toImmutableList());
   }
 
+  private String getInvocationId(HttpHeaders header) {
+    String apiClientHeader = (String) header.get(InvocationIdInterceptor.GOOG_API_CLIENT);
+    int beginIndex = apiClientHeader.indexOf(InvocationIdInterceptor.GCCL_INVOCATION_ID_PREFIX);
+    if (beginIndex >= 0) {
+      beginIndex = beginIndex + InvocationIdInterceptor.GCCL_INVOCATION_ID_PREFIX.length();
```

Comment:
it would be helpful if we can provide a sample header as comment.

Reply:
Sure.